third_party/airflow/armada/operators/armada.py

```diff
@@ -131,3 +131,23 @@ def _get_lookout_url(self, job_id: str) -> str:
         if self.lookout_url_template is None:
             return ""
         return self.lookout_url_template.replace("<job_id>", job_id)
+
+    def on_kill(self) -> None:
+        """
+        Stops the JobService from listening to the JobSet and cancels the jobs.
+
+        :return: None
+        """
+        try:
+            if self.job_set_id and self.queue:
```

Comment:
do we need an `else` clause here? or, if it's always true, maybe we don't need the `if`?

Reply:
If it's always true that  self.job_set_id and self.queue always have value whenever on_kill() called  then we can remove the 'if' statement !!

Reply:
I'm not super familiar with this part of the code base, but I'd suggest follow pattern from elsewhere, eg: https://github.com/armadaproject/armada/blob/7508265ddfaabd17c79ba69af2f1e7ecae288de3/third_party/airflow/armada/operators/armada.py#LL106C1-L109C71

That is, if we lack job_set_id and queue for some reason, raise an AirflowException. And if we catch an exception canceling the job, also raise an AirflowException.